spec/Staking.md

```diff
@@ -47,16 +47,40 @@ makeKey(...fragments) = rlp([
     The RLP-encoded non-empty list should be sorted by a `delegatee` in ascending order, and every `delegatee` should be unique.
 
-## Pending Revocations
+## List of Candidates
 
-  * State Key: `makeKey("Revocations")`
+  * State Key: `makeKey("Candidates")`
+  * Value: `rlp(list of [pubkey, deposit, nominations_ends_at, metadata])`
 
-  * Value: `rlp(list of [delegator, delegatee, endTime, quantity])`
+    The `pubkey` is a public key of a self-nominated candidate. The `deposit` is a `u64` amount of CCS deposited by the candidate and `nomination_ends_at` is a `u64` term id when the entry is expired. The `metadata` is a `bytes` that can store a short amount of data that express or advertise themselves. The order of it is constantly changed as the candidates send self-nominate transaction and the term finishes. See the 'Candidate prioritizing' section of [dynamic validator](./Dynamic-Validator.md#Candidate-prioritizing).
 
-    A `delegator` is an `AccountId` who has delegated CCS to a `delegatee`, which is also an `AccountId`.
-    `endTime` is a Unix time in UTC that specifies when the revocation will finally end. `quantity` is the `u64` amount of CCC that is going to be revoked.
-    The RLP-encoded non-empty list should be sorted by `endTime` in ascending order.
-    When multiple revocations have the same `endTime`, then the revocation created earlier (a block number is smaller, a transaction index is smaller) must have a smaller index.
+## List of jailed accounts
+
+  * State Key: `makeKey("Jailed")`
+  * Value: `rlp(list of [account, deposit, custody_until, released_at])`
+
+    The `account` is an `AccountId`, and the `deposit` is a `u64` amount of CCS deposited before the candidate is jailed. A jailed candidate can self-nominate and be removed from the list after the term id is greater or equal than a `u64` value of `custody_until`, and it is automatically removed when the term id is greater or equal than a `u64` value of `released_at`. The RLP-encoded non-empty list should be sorted by an `account` in ascending order, and every `account` should be unique.
+
+## List of banned accounts
+
+  * State Key: `makeKey("Banned")`
+  * Value: `rlp(list of account)`
+
+    The `account` is an `AccountId`.
+
+## Current validator set
+
+  * State Key: `makeKey("Validators")`
+  * Value: `rlp(list of [weight, delegation, deposit, pubkey])`
+
+    See 'How to update validators' section in [dynamic validator](./Dynamic-Validator.md#How-to-update-validators).
+
+## Intermediate rewards
+
+  * State Key: `makeKey("IntermediateRewards")`
+  * Value: `rlp([list of [account,quantity], list of [account,quantity]])`
+
+    The `address` is an `AccountId`, and `quantity` is a `u64` value of CCC. The value is the RLP encoded list of two lists. The first list is the rewards of previous term, and the second list is the rewards of current term. Each list is sorted by the `account` in ascending order, and every `account` in a list should be unique.
```

Comment:
The `address` is an `AccountId`, and `quantity` is a `u64` value of CCC. The value is an RLP encoded list of two lists. The first list is the rewards of the previous term, and the second list is the rewards of the current term. Each list is sorted by `account` in ascending order, and every `account` in a list should be unique.